microservices/notification-service/src/member/member.event-subscriber.ts

```diff
@@ -0,0 +1,31 @@
+import {Injectable, Logger} from "@nestjs/common";
+import type {EventResponse} from "@sprocketbot/common";
+import {
+    EventsService, EventTopic,
+} from "@sprocketbot/common";
+
+import {MemberService} from "./member.service";
+
+@Injectable()
+export class MemberEventSubscriber {
+    private readonly logger = new Logger(MemberEventSubscriber.name);
+
+    constructor(
+        private readonly eventsService: EventsService,
+        private readonly memberService: MemberService,
+    ) {}
+
+    async onApplicationBootstrap(): Promise<void> {
+        await this.eventsService.subscribe(EventTopic.MemberRestrictionCreated, false).then(obs => {
+            // eslint-disable-next-line @typescript-eslint/no-misused-promises
+            obs.subscribe(this.onMemberRestrictionCreated);
+        });
+    }
+
+    onMemberRestrictionCreated = async (d: EventResponse<EventTopic.MemberRestrictionCreated>): Promise<void> => {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        if (d.topic !== EventTopic.MemberRestrictionCreated) return;
+
```

Comment:
Add check for restriction type here instead of service call?

Reply:
I think it makes sense where it is, because the method is specifically named `sendQueueBanNotification`, so it has queue ban in there.